metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -899,6 +899,77 @@ def _extract_lineages(self):
         return
 
 
+@dataclass
+class ConfluentS3SinkConnector:
+    connector_manifest: ConnectorManifest
+
+    def __init__(
+        self, connector_manifest: ConnectorManifest, report: KafkaConnectSourceReport
+    ) -> None:
+        self.connector_manifest = connector_manifest
+        self.report = report
+        self._extract_lineages()
+
+    @dataclass
+    class S3SinkParser:
+        target_platform: str
+        bucket: str
+        topics_dir: str
+        topics: List[str]
+
+    def _get_parser(self, connector_manifest: ConnectorManifest) -> S3SinkParser:
+        # https://docs.confluent.io/kafka-connectors/s3-sink/current/configuration_options.html#s3
+        bucket = connector_manifest.config.get("s3.bucket.name")
+        if not bucket:
+            raise ValueError(f"Could not find 's3.bucket.name' in connector configuration {self.connector_manifest.name}")
```

Comment:
Although this is unlikely in correctly configured connector, we should catch such exceptions and fail gracefully only within scope of this connector. Maybe add a try..except inside ConfluentS3SinkConnector._extract_lineages ?

Reply:
What is the expected behaviour of the `KafkaConnect.get_workunits_internal(..)` method, if there is an error processing any of the connector configs? Should it just report a failure/warning, ignore the connector and return `MetadataWorkUnit`s for others? If that’s the case, we could just invoke `report.report_failure` here, instead of throwing an exception.

Although, there are examples where `get_workunits_internal()` does both eg https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/pulsar.py#L185. 

Thoughts?

Reply:
We prefer reporting a warning for that scope(failing connector config here) and continue processing others.

For reporting a warning, you can choose to enclose connector scoped code within try..except and raise as many exceptions within as required as long as you catch them in that scope and report them without causing fatal exceptions outside that scope.